### PR TITLE
Make flake8 happier

### PR DESCRIPTION
**Pull request by eumiro**
This makes `flake8` much happier about the `tables/` directory.

As discussed in #867, we should set and check some coding standards, such as PEP8. For this, first the whole codebase should pass the `flake8` test and then we can have some automatic check for new PR in order not to contaminate the code.

This PR solves most of the issues reported by

```
python3 -m flake8 --extend-ignore=D,N --statistics --count tables
```

Before this PR, there were 1268 problems. After this PR we are down to some `E501` (line too long) in doctest and template that probably can be reformatted and the rest are mostly caused by wildcard imports to be investigated and normalized to `from x import y`.

All the individual commits can be squashed to a single one afterwards.

### CODE REVIEW

**Comment by eumiro** [comment]
Finally it works. Only Travis is `Unable to connect to ports.ubuntu.com:http: [IP: 91.189.88.142 80]`, but the rest is satisfied. This is what left:

```
20    E402 module level import not at top of file
9     E501 line too long (80 > 79 characters)
9     F401 '.utilsextension.blosc_compressor_list' imported but unused
3     F403 'from .atom import *' used; unable to detect undefined names
69    F405 'Complex32Col' may be undefined, or defined from star imports: .atom, .description, .exceptions
11    F821 undefined name '_AssertWarnsContext'
121
```

but will have to be analyzed and fixed in a new PR.

File: tables/atom.py
Comment (avalentino) [inline tables/atom.py:370]: OK in principle thin could impact doctests but this particular test is SKIPed and doctests seems to be broken in general (see #874).
Probably it is better to include the change and think to broken doctests in a separate PR

**Review by avalentino — APPROVED** [review]
LGTM, just please leave @FrancescAlted the time to give feedback about emacs configuration comments before merging

**Comment by avalentino** [comment]
> All the individual commits can be squashed to a single one afterwards.

for me it is not necessary to squash but please do what you feel is better

File: tables/attributeset.py
Comment (eumiro) [reply tables/attributeset.py:703]: I have removed them because `flake8` does not like double `##` as comment. But actually I believe they do not really belong to the project, since any random editor/IDE could leave some traces that would render the source code confusing.

Another thing we could consider removing is the long header of *some* files (with the name of the author(s), some date around 2003, license, etc.). Do we need these legally? There's a license with the source code, and there's the whole git history that tells much more about the author(s) and changes).

And maybe you could also just grep and hopefully delete all commented out code in the repository. I had to touch quite a lot of it in this PR without being sure whether this would be even needed. You remember much more about the code than I do.

File: tables/atom.py
Comment (eumiro) [reply tables/atom.py:370]: Oh, right, didn't take the doctests into account until you posted #874. How do actually doctests work when the line is too long for `flake8`?

File: tables/attributeset.py
Comment (FrancescAlted) [reply tables/attributeset.py:703]: Please feel free to remove all the legacy emacs metainfo.

Also, I am fine with removing the copyright headers. Using a reference to a general LICENSE file (conveniently updated to include new maintainers :-) should be enough.

File: tables/atom.py
Comment (avalentino) [reply tables/atom.py:370]: In my understanding doctest compares the output verbatim but I'm not sure what are indentation rules.
In any case there is the ``NORMALIZE_WHITESPACE`` directive that could help in this cases.

File: tables/attributeset.py
Comment (avalentino) [reply tables/attributeset.py:703]: Regarding the header I don't think we are forced to add the copyright statement in each file )but I could be wrong).
In any case before removing it IMHO it is better to ask to the person mentioned in the header itself.

Regarding commented code probably most of it is obsolete but in many cases there could be something useful for debugging/future enhancements.
IMHO a case by case review is necessary.

Comment (avalentino) [reply tables/attributeset.py:703]: ups, sorry @FrancescAlted I have not seen you comment before posting mine

**Pull request merged**
